src/components/fields/_BasicCell/BasicCellContextMenuActions.tsx

```diff
@@ -5,56 +5,63 @@ import CopyCells from "@src/assets/icons/CopyCells";
 import Paste from "@mui/icons-material/ContentPaste";
 
 import { useProjectContext } from "@src/contexts/ProjectContext";
+import { useSnackbar } from "notistack";
+import { SelectedCell } from "@src/atoms/ContextMenu";
 
-export default function BasicContextMenuActions() {
-  const { contextMenuRef, tableState, deleteCell, updateCell } =
-    useProjectContext();
+export interface IContextMenuActions {
+  label: string;
+  icon: React.ReactNode;
+  onClick: () => void;
+}
+
+export default function BasicContextMenuActions(
+  selectedCell: SelectedCell,
+  reset: () => void | Promise<void>
+): IContextMenuActions[] {
+  const { tableState, deleteCell, updateCell } = useProjectContext();
+  const { enqueueSnackbar } = useSnackbar();
   const columns = tableState?.columns;
   const rows = tableState?.rows;
-  const selectedRowIndex = contextMenuRef?.current?.selectedCell
-    .rowIndex as number;
-  const selectedColIndex = contextMenuRef?.current?.selectedCell?.colIndex;
+  const selectedRowIndex = selectedCell.rowIndex as number;
+  const selectedColIndex = selectedCell?.colIndex;
   const selectedCol = _find(columns, { index: selectedColIndex });
   const selectedRow = rows?.[selectedRowIndex];
+  const cell = selectedRow?.[selectedCol.key];
 
-  const handleClose = () => {
-    contextMenuRef?.current?.setSelectedCell(null);
-    contextMenuRef?.current?.setAnchorEl(null);
-  };
-
-  const handleCopy = () => {
-    const cell = selectedRow?.[selectedCol.key];
-    // const onFail = () => console.log("Fail to copy");
-    // const onSuccess = () => console.log("Save to clipboard successful");
-    // const copy =
-    navigator.clipboard.writeText(JSON.stringify(cell));
-    // copy.then(onSuccess, onFail);
+  const handleClose = async () => await reset?.();
 
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(JSON.stringify(cell));
+      enqueueSnackbar("Copied to clipboard", { variant: "success" });
+    } catch (error) {
+      enqueueSnackbar("Failed to copy", { variant: "error" });
+    }
     handleClose();
   };
 
-  const handleCut = () => {
-    const cell = selectedRow?.[selectedCol.key];
+  const handleCut = async () => {
     const notUndefined = Boolean(typeof cell !== "undefined");
```

Comment:
```suggestion
    const notUndefined = cell !== undefined;
```

The `!==` operator should always return a boolean, so it’s not necessary to use the `Boolean` object for conversion here. You can also compare directly to `undefined` without using `typeof`